kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -153,6 +153,8 @@ object Async:
     /** Races multiple computations and returns the result of the first to complete. When one computation completes, all other computations
       * are interrupted.
       *
+      * WARNING: Executes all computations in parallel without bounds. Use with caution on large sequences to avoid resource exhaustion.
```

Comment:
I've added this warning to `race` and `gather` methods since they don't have a parallelism limit.